cascade-merge/action.yml

```diff
@@ -0,0 +1,200 @@
+            echo " === $msg OK ==="
+        done
+      shell: bash
+
+#   waiting for slack secrets to be shared as organization secrets:
+#    https://ledgerhq.atlassian.net/servicedesk/customer/portal/14/ACCESS-3920?created=true
+#    - name: Notify failing merge
+#      if: failure()
+#      uses: LedgerHQ/actions/slack-notification@main
+#      env:
+#        SLACK_WEBHOOK_URL: ${{ inputs.slack-webhook-url }}
+#        SLACK_API_TOKEN: ${{ inputs.slack-api-token }}
+#        SLACK_CUSTOM_PAYLOAD: |-
+#          {
+#            "blocks": [
+#              {
+#                "type": "header",
+#                "text": {
+#                  "type": "plain_text",
+#                  "text": ":boom: Merge conflict :boom:",
+#                  "emoji": true
+#                }
+#              },
+#              {
+#                "type": "section",
+#                "text": {
+#                  "type": "mrkdwn",
+#                  "text": "<@SLACK_MEMBER_ID> please fix it :iamwatchingyou:"
+#                }
+#              },
+#              {
+#                "type": "actions",
+#                "elements": [
+#                  {
+#                    "type": "button",
+#                    "text": {
+#                      "type": "plain_text",
+#                      "text": ":github: Github build",
+#                      "emoji": true
+#                    },
+#                    "url": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+#                  }
+#                ]
+#              },
+#              {
+#                "type": "divider"
+#              }
+#            ]
+#          }
+#
+
+    # checkout again to clean all local merges done in previous step
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0  # fetch all history so we can push to other branches
+
+    - name: Cascade merge
+      if: "! contains(env.COMMIT_MSG, '[skip cascade merge]')"  # continue only if the message does not contain '[skip cascade merge]'
+      run: |
+        # Set Bash to print all commands and stop executing on any error
+        set +e
+
+        # Define helper functions to echo messages in color
+        function echo_green {
+            echo -e "\e[32m$1\e[0m"
+        }
+        function echo_red {
+            echo -e "\e[31m$1\e[0m"
+        }
+        
+        # Define function to handle PR creation
+        function handle_pr_creation {
+          echo_green "Creating PR to cascade merge from $1 to $2"
+          NEW_PR_URL=$(gh pr create --title "Cascade merge from $1 to $2" --body "Automated cascade merge" --base $2 --head $1 --repo ${{ github.repository }})
+          PR_CREATION_STATUS=$?
+
+          if [[ $PR_CREATION_STATUS -ne 0 ]]; then
+            echo_red "PR creation failed."
+            exit 1
+          else
+            echo_green "PR is created and ready for review: $NEW_PR_URL"
```

Comment:
maybe take advantage of github summaries so we don't have to go into the job logs to find the result?
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
this will create a nice summary in the workflow page, without having to click on the job and then check the steps logs individually
you can add this everywhere you have a result

Reply:
applied 
![image](https://github.com/LedgerHQ/actions/assets/96146489/08b7aeac-f6f7-4bfc-bcad-77559ffd5962)